interface/modules/custom_modules/oe-module-faxsms/messageUI.php

```diff
@@ -0,0 +1,647 @@
+        <nav class="navbar navbar-expand-xl navbar-light bg-light">
+            <div class="container">
+                <a class="navbar-brand" href="#"><h4><?php echo $title; ?></h4></a>
+                <button type="button" class="bg-primary navbar-toggler mr-auto" data-toggle="collapse" data-target="#nav-header-collapse">
+                    <span class="navbar-toggler-icon"></span>
+                </button>
+                <div class="collapse navbar-collapse" id="nav-header-collapse">
+                    <form class="navbar-form navbar-left form-inline" method="GET" role="search">
+                        <div class="form-group">
+                            <label class="mx-1 font-weight-bolder" for="formdate"><?php echo xlt('Activities From Date') ?>:</label>
+                            <input type="text" id="fromdate" name="fromdate" class="form-control input-sm datepicker" placeholder="YYYY-MM-DD" value=''>
+                        </div>
+                        <div class="form-group">
+                            <label class="mx-1 font-weight-bolder" for="todate"><?php echo xlt('To Date') ?>:</label>
+                            <input type="text" id="todate" name="todate" class="form-control input-sm datepicker" placeholder="YYYY-MM-DD" value=''>
+                        </div>
+                        <div class="form-group">
+                            <button type="button" class="btn btn-primary btn-search" onclick="retrieveMsgs(event,this)" title="<?php echo xla('Click to get current history.') ?>"></button>
+                        </div>
+                        <!-- manual login oauth2 RC -->
+                        <div class="form-group ringcentral">
+                            <button id="loginButton" onclick="location.reload()" class="btn btn-danger d-none">Log In<i class="fa fa-sign-in-alt ml-2"></i></button>
+                        </div>
+                    </form>
+                    <div class="nav-item dropdown ml-auto">
+                        <button class="btn btn-primary dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
+                            <?php echo xlt('Actions'); ?><span class="caret"></span>
+                        </button>
+                        <div class="dropdown-menu" role="menu">
+                            <a class="dropdown-item" href="#" onclick="doSetup(event)"><?php echo xlt('Account Credentials'); ?></a>
+                            <?php if ($serviceType == 'sms') { ?>
+                                <a class="dropdown-item" href="#" onclick="popNotify('', './library/rc_sms_notification.php?dryrun=1&site=<?php echo $_SESSION['site_id'] ?>')"><?php echo xlt('Test SMS Reminders'); ?></a>
+                                <a class="dropdown-item" href="#" onclick="popNotify('live', './library/rc_sms_notification.php?site=<?php echo $_SESSION['site_id'] ?>')"><?php echo xlt('Send SMS Reminders'); ?></a>
+                            <?php } ?>
```

Comment:
attr_url() around $_SESSION['site_id'] in 2 lines above

Reply:
can't use js_url here. using attr

Reply:
try this:
```
<a class="dropdown-item" href="#" onclick="popNotify('', './library/rc_sms_notification.php?dryrun=1&site=' + <?php echo attr(js_url($_SESSION['site_id'])) ?>)"><?php echo xlt('Test SMS Reminders'); ?></a>
                                <a class="dropdown-item" href="#" onclick="popNotify('live', './library/rc_sms_notification.php?site=' + <?php echo attr(js_url($_SESSION['site_id'])) ?>)"><?php echo xlt('Send SMS Reminders'); ?></a>
```
first time have had to use three escaping at one time :)